youtube_transcript_api/test/test_cli.py

```diff
@@ -50,14 +77,26 @@ def test_argument_parsing__languages(self):
         self.assertEqual(parsed_args.json, False)
         self.assertEqual(parsed_args.languages, ['de', 'en'])
 
+    def test_argument_parsing__proxies(self):
+        parsed_args = YouTubeTranscriptCli(
+            'v1 v2 --http-proxy http://user:pass@domain:port'.split()
+        )._parse_args()
+        self.assertEqual(parsed_args.http_proxy, 'http://user:pass@domain:port')
+
+        parsed_args = YouTubeTranscriptCli(
+            'v1 v2 --https-proxy https://user:pass@domain:port'.split()
+        )._parse_args()
+        self.assertEqual(parsed_args.https_proxy, 'https://user:pass@domain:port')
+
     def test_run(self):
```

Comment:
You'll also have to run a version of this test with proxies being set, to get full coverage

Reply:
Thanks for letting me know!

Reply:
So I just moved the coverage a tiny bit, perhaps I misunderstood what you meant by test with proxies being set?

Reply:
Yeah, you might have misunderstood me here :)

During the build coveralls checks if all branches of the code are covered by the tests which are run. You can see the results here: https://coveralls.io/builds/22302114/source?filename=youtube_transcript_api%2F_cli.py#L19

As you can see the part of the `run()` method which is only called when a proxy is set, is not executed during any of the tests. Since the tests you added only test `_parse_args()` but not `run()` they won't cover this line. To do so you'll need to add the following test:

```python
def test_run__proxies(self):
    YouTubeTranscriptApi.get_transcripts = MagicMock(return_value=([], []))
    YouTubeTranscriptCli('v1 v2 --languages de en --http-proxy http://user:pass@domain:port --https-proxy https://user:pass@domain:port'.split()).run()

    YouTubeTranscriptApi.get_transcripts.assert_called_once_with(
        ['v1', 'v2'],
        languages=['de', 'en'],
        continue_after_error=True,
        proxies={'http': 'http://user:pass@domain:port', 'https': 'https://user:pass@domain:port'}
    )
```